### PR TITLE
MISO schema update

### PR DESCRIPTION
**Pull request by djcooke**
This PR is for the GLT-1519 change only. Will merge GLT-1412 (https://github.com/oicr-gsi/pinery/pull/21) before this

### CODE REVIEW

File: pinery-lims-miso/src/main/java/ca/on/oicr/pinery/lims/miso/MisoClient.java [hunk 1]
```diff
@@ -318,8 +317,17 @@
       "        INNER JOIN Indices ON Indices.indexId = Library_Index.index_indexId\n" + 
       "                WHERE position = 2\n" + 
       "        ) bc2 ON bc2.library_libraryId = l.libraryId\n" + 
-      "LEFT JOIN BoxPosition pos ON pos.boxPositionId = l.boxPositionId\n" + 
+      "LEFT JOIN BoxPosition pos ON pos.targetId = l.libraryId\n" + 
+      "        AND pos.targetType = 'L'\n" +
```
Comment (callunity) [inline pinery-lims-miso/src/main/java/ca/on/oicr/pinery/lims/miso/MisoClient.java]: I know that these are `S` and `L` in the db, but @djcooke [requested](https://github.com/TGAC/miso-lims/pull/676) that @apmasell change the targetType to "something more descriptive like 'Sample' and 'SampleTissue'" and the change was made in the BoxImpl but maybe not in the db migration script. Will this cause an issue here?

Comment (djcooke) [inline pinery-lims-miso/src/main/java/ca/on/oicr/pinery/lims/miso/MisoClient.java]: I thought I had made that request, and I thought it had been done, but then there were S's and L's :confused:

Comment (apmasell) [inline pinery-lims-miso/src/main/java/ca/on/oicr/pinery/lims/miso/MisoClient.java]: Oh, this should be `LIKE 'L%'` and similar `LIKE 'S%'` for samples.

Comment (djcooke) [inline pinery-lims-miso/src/main/java/ca/on/oicr/pinery/lims/miso/MisoClient.java]: This is what my database looks like:
```
mysql> SELECT * FROM BoxPosition;
+-------+--------------+----------+------------+----------+
| boxId | lastModifier | targetId | targetType | position |
+-------+--------------+----------+------------+----------+
|     1 |            1 |        8 | P          | A01      |
|     3 |            1 |        1 | L          | A01      |
|     3 |            1 |        2 | P          | B01      |
|     4 |            1 |      230 | S          | A01      |
|     4 |            1 |      231 | S          | B01      |
|     4 |            1 |      232 | S          | C01      |
```

**Comment by djcooke** [comment]
Updated to match fixes in https://github.com/TGAC/miso-lims/pull/704

**Pull request merged**